Fix part of #10306 Added "async" keyword to asynchronous functions that return a promise to question-backend-api.service.ts

abhishekalbert opened the pull request:
Add "async" keyword to asynchronous functions that return a promise

## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #10306.
2. This PR does the following: Added "async" keyword to asynchronous functions that return a promise to question-backend-api.service.ts (Path: core/templates/domain/question/question-backend-api.service.ts)

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes made in this PR work correctly.
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Oppiabot can assign anyone for review/help if you leave a comment like the following: "{{Question/comment}} @{{reviewer_username}} PTAL"
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-your-build-fails).

abhishekalbert committed:
Resolved issue #10306

Add "async" keyword to asynchronous functions that return a promise

kevintab95 commented on core/templates/domain/question/question-backend-api.service.ts:
```diff
@@ -181,7 +181,7 @@ export class QuestionBackendApiService {
    * of questions requested.
    */
-  fetchQuestions(
+ async fetchQuestionsAsync(
```
Hm, do you need to make changes in all the places where this method is called? Ditto below.

kevintab95 reviewed (changes requested):
Thanks @abhishekalbert! left a comment. PTAL!

kevintab95 commented:
Hi @abhishekalbert, the PR title needs to explain briefly what this PR is about. Please see the first item in the checklist `The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)`.

abhishekalbert committed:
Fix part of #10306

Added "async" keyword to asynchronous functions that return a promise to question-backend-api.service.ts (Path: core/templates/domain/question/question-backend-api.service.ts)

abhishekalbert commented:
Hello, sir @kevintab95 
Thanks for reviewing. As you told, I updated the PR please let me know if anything wrong.

abhishekalbert committed:
Update question-backend-api.service.ts

sajalasati commented:
> Hello, sir @kevintab95
> Thanks for reviewing. As you told, I updated the PR please let me know if anything wrong.

Hey @abhishekalbert! If you have addressed all the changes, please reply to each of the comment, and leave a `@reviewer PTAL` for the reviewer. Currently the PR wasn't assigned to @kevintab95 that's why he hasn't reviewed. Make sure the PR is always assigned to the right person.

sajalasati commented:
> Hello, sir @kevintab95
> Thanks for reviewing. As you told, I updated the PR please let me know if anything wrong.

PTAL @kevintab95!

abhishekalbert commented:
@kevintab95 and @sajalasati because of some internal PC error my progress with this PR get blocked so yesterday I created a new PR #12188 and also made some progress as compared to this one right now I'm getting issues with backend test so can please look at #12188 and help me figure out how to resolve it.  
Thank you.

kevintab95 commented:
Superseded by https://github.com/oppia/oppia/pull/12188. Closing.
